Design.md

```diff
@@ -229,6 +230,14 @@ This API will enable or disable user on Vault.
 ### Create S3 Credential
 This API creates S3 credential for the user.
 1. `generate-access-key` api will be called using assumed role credentials.
+1. Encode the `secret-key` using Base64 encoder and store the encoded `secret-key` on Redis Cache with key formatted as `<Username>__<AccessKeyID>`.
+
+_Notes:_
+* _Redis cache will always be rolling deployed._
+* _If OSE restarts, all the access keys created before the `ose-restart-time` needs to be deleted from Vault. A script will be provided to be run manually only when OSE is restarted._ 
```

Comment:
> all the access keys created before the `ose-restart-time`

vCloud Director Tenant Users can generate access keys programmatically using IAM API. If your script deletes all the access keys in Vault created  before `ose-restart-time`, you will also delete the keys generated programmatically. 

Your script must affect only the keys created by OSE/OSIS.

Reply:
As @vrancurel suggested above, we actually do not need to clear the keys but need to log the `access keys` created by OSIS. So later if we want to delete these keys created before certain date, these `access-key-ids` will be used.

Reply:
I will add a note regarding the logging for OSIS created access keys

Reply:
Addressed in 9e7c162b06f6f686a847a767ce61219fd4caaf9f